tests/gendir

```diff
@@ -0,0 +1,195 @@
+#!/usr/bin/python3
```

Comment:
This repository is now up to 3 distinct languages  (C, shell script, and Python).  Personally, I would vote that eventually of using Python and shell script for tests we use Rust.  I've had decent success with this, see e.g. https://github.com/ostreedev/ostree/blob/878d601665fd15781adf774d99dd86e9a172cb92/tests/inst/src/destructive.rs#L477 shows how Rust's macros can let one execute shell commands ([xshell crate](https://docs.rs/xshell/latest/xshell/)) with near-equal ergonomics to bash, but more safety than Python.

Reply:
I love rust for production code, but I'm not sure how well it would fit with code that is a bit more throwaway and like tests. Seems like it may raise the bar for people to add new testcases. I'd be willing to be convinced though.